### ISSUE TITLE
`Deleted Report` is showing in the LHN for the shared user

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#42326]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.74-4 
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @miljakljajic 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1715854992799659

## Action Performed:
1. Create a workspace as user A and invite user B as an employee in NewDot
2.  Do not toggle "Add approvals" in More features
3. Login as user B in OD and submit the expense and CC the report to user C
4. Observe the LHN of the user C
## Expected Result:
User C shouldn't see the "deleted report" in the LHN
## Actual Result:
User C is seeing "deleted report" in the LHN.
## Workaround:
unknown 
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/693d00e0-eb47-4408-9467-3474dc230b43

![image (3)](https://github.com/Expensify/App/assets/38435837/3e99653b-576d-4a1e-ba13-4006f6c0d17b)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by miljakljajic** [comment]
I should add also that now these items can't be cleared, and don't lead anywhere: 
<img width="1436" alt="image" src="https://github.com/Expensify/App/assets/28715117/bcc4baec-f566-4f69-a050-0ed6152a481d">

**Comment by stephanieelliott** [comment]
I think this can be fixed externally, added labels to get some proposals!

**Comment by stephanieelliott** [comment]
Still waiting on some proposals. In the meantime, @miljakljajic is this issue still occurring for you?

**Comment by stephanieelliott** [comment]
... this is not overdue, I just updated yesterday 

> Still waiting on some proposals. In the meantime, @miljakljajic is this issue still occurring for you?

**Comment by rojiphil** [comment]
@stephanieelliott @miljakljajic I could reproduce the issue with one additional step between (3) and (4) 
3. Login as user B in OD and submit the expense and CC the report to user C
(3a) Open the submitted report using deep link on user C login. 
4. Observe the LHN of the user C

Actual Behaviour: [Deleted Report] is shown temporarily until report is loaded
Expected Behaviour: [Deleted Report] should not be shown temporarily. But, then, what should be shown?

https://github.com/Expensify/App/assets/3004659/261589dd-144a-4104-a215-0bc8055bfdb8

**Comment by stephanieelliott** [comment]
Ah ok, thanks @rojiphil! Ok so then the full repro steps would be: 

1. In NewDot: create a workspace as User A and invite User B as an employee 
2. In the workspace, ensure that  "Add approvals" in More features is not toggled on. 
3. Login as User B in OldDot and submit the expense. CC the report to user C
4. As User C, open the report via the deep link; observe the LHN for problem behavior 

**Actual Behaviour:** [Deleted Report] is shown temporarily until report is loaded
**Expected Behaviour: [**Deleted Report] should not be shown temporarily. But, then, what should be shown?

Alright, so it sounds like this only happens under the condition where the report is submitted via OldDot **and** the report is opened via the deep link. With that being the case this sounds pretty edge case -- once someone is migrated they will be operating in ND exclusively so this shouldn't occur. 

Seeing as `Deleted Report` is only being shown temporarily until the report loads, I'm inclined to close this. But I will wait for other voices to chime in!

**Comment by rojiphil** [comment]
> Seeing as `Deleted Report` is only being shown temporarily until the report loads, I'm inclined to close this. But I will wait for other voices to chime in!

Yep. Sounds pretty edge case to me too and am also inclined to close.
